lookup/lookup.js

```diff
@@ -59,17 +61,44 @@ async function main (params) {
   };
 }
 
-async function lookup (args) {
+function lookup (args) {
   const agreements = args.agreements;
-  let usernames = [];
-  await Promise.all(agreements.map(async function (agreement) {
-    const agreementUsers = await lookupSingleAgreement(args, agreement);
-    usernames = Array.from(new Set(usernames.concat(agreementUsers)));
-  }));
-  return usernames;
+  const username = args.username.toLowerCase();
+  const promiseMatrix = agreements.map(agreement => lookupAndParseAgreement(args, agreement));
+  const [responsePromises, usernamesPromises] = transpose(promiseMatrix);
+
+  // A new Promise that resolves to either
+  // 1. Username from the first Sign API call that includes the username
+  // 2. Empty array if none of the Sign API calls includes the username
+  // Promise is rejected if any of the Sign API calls fail before resolution
+  // If the username is found, all requests are aborted (it's okay to abort completed requests)
+  return new Promise((resolve, reject) => {
+    Promise.all(usernamesPromises.map(promise => {
+      return promise.then(agreementUsers => {
+        if (agreementUsers.map(u => u.toLowerCase()).includes(username)) {
+          resolve([username]);
```

Comment:
I considered refactoring this to return a boolean, but I held off:
1. There's already quite a bit of refactoring in this PR
2. Hypothetical usecase: looking up multiple usernames (e.g. PR has commits from multiple authors, all of which need to have signed the CLA)